pkg/server/status_test.go

```diff
@@ -662,9 +665,158 @@ func TestHotRangesPayloadMultitenant(t *testing.T) {
 			return err
 		}
 
-		if !hasDescriptorTable(resp) {
+		if !responseHasTable(resp, "descriptor") {
 			return errors.New("waiting for hot ranges to be collected")
 		}
 		return nil
 	})
 }
+
+func justIds(ranges []*serverpb.HotRangesResponseV2_HotRange) []int {
+	ids := []int{}
+	for i := range len(ranges) {
+		ids = append(ids, int(ranges[i].RangeID))
+	}
+	slices.Sort(ids)
+	return ids
+}
+
+func assertRangesEqualish(t *testing.T, a, b []*serverpb.HotRangesResponseV2_HotRange) {
+	require.Equal(t, justIds(a), justIds(b))
+}
+
+func assertRangesNotEqualish(t *testing.T, a, b []*serverpb.HotRangesResponseV2_HotRange) {
+	require.NotEqual(t, justIds(a), justIds(b))
```

Comment:
nit: avoid single line helpers. they harm readability. Also, mixing `require` and `assert` naming should be avoided. They have different meanings. `assert` will continue execution while `require` halts.

Also, helpers should call `t.Helper()` so that failure reporting is in the test function, and not in here. Makes it a bit tougher to diagnose failures.